File: LeetCode/jump-game.cpp
// https://leetcode.com/problems/jump-game/
class Solution
{
public:
    bool canJump(vector<int> &nums)
    {
        int furthest = 0;
        for (int i = 0; i < nums.size() && i <= furthest; ++i)
        {
            furthest = max(furthest, i + nums[i]);
        }
        return furthest >= nums.size() - 1 ? true : false;
    }
};

File: LeetCode/jump-game.java
// https://leetcode.com/problems/jump-game/

class Solution {
  public boolean canJump(int[] nums) {
    int furthest = 0;

    for (int i = 0; i < nums.length && i <= furthest; ++i) {
      furthest = Math.max(furthest, i + nums[i]);
    }

    return furthest >= nums.length - 1 ? true : false;
  }
}